types/vote.go

```diff
@@ -316,7 +316,7 @@ func (vote *Vote) ValidateBasic() error {
 	// We should only ever see vote extensions in non-nil precommits, otherwise
 	// this is a violation of the specification.
 	// https://github.com/tendermint/tendermint/issues/8487
-	if vote.Type != tmproto.PrecommitType || (vote.Type == tmproto.PrecommitType && len(vote.BlockID.Hash) == 0) {
+	if vote.Type != tmproto.PrecommitType || (vote.Type == tmproto.PrecommitType && vote.BlockID.IsZero()) {
 		if len(vote.Extension) > 0 {
 			return errors.New("unexpected vote extension")
```

Comment:
include the vote.Type in the error to allow figuring which case it was.

Reply:
Done. Also included "isNil"